src/hotspot/cpu/x86/x86.ad

```diff
@@ -1782,20 +1782,6 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
         return false; // Implementation limitation
       }
       break;
-    case Op_VectorCastB2X:
-    case Op_VectorCastS2X:
-    case Op_VectorCastI2X:
-      if (bt != T_DOUBLE && size_in_bits == 256 && UseAVX < 2) {
-        return false;
-      }
-      break;
-    case Op_VectorCastL2X:
-      if (is_integral_type(bt) && size_in_bits == 256 && UseAVX < 2) {
```

Comment:
No new code for `VectorCastL2X` to integral types? Why?

Reply:
This is also redundant, the only long vector that can appear in AVX1 is <= 128-bit, so we can't cast to 256-bit integral vectors here.